src/bsp/pc-linux/src/bsp_console.c

```diff
@@ -0,0 +1,102 @@
+/******************************************************************************
+** File:  bsp_console.c
+**
+**
+**      This is governed by the NASA Open Source Agreement and may be used,
+**      distributed and modified only pursuant to the terms of that agreement.
+**
+**      Copyright (c) 2004-2006, United States government as represented by the
+**      administrator of the National Aeronautics Space Administration.
+**      All rights reserved.
+**
+**
+** Purpose:
+**   OSAL BSP debug console abstraction
+**
+******************************************************************************/
+
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include "pclinux_bsp_internal.h"
+#include "bsp-impl.h"
+
+/*----------------------------------------------------------------
+   OS_BSP_ExecTput()
+
+   Helper function: Use the system "tput" utility to set the given
+   console capability.
+
+   This uses a fork/exec to invoke the external command which outputs
+   the control sequence directly to the controlling terminal.
+
+   It is assumed that this will only be used during debug/testing.
+   Otherwise it would be preferable to cache the control strings to
+   avoid repetitive fork/exec operations.
+ ------------------------------------------------------------------*/
+static void OS_BSP_ExecTput(const char *cap, const char *param)
+{
+    pid_t cpid;
+    int   status;
+
+    cpid = fork();
+    if (cpid < 0)
+    {
+        return;
+    }
+    if (cpid == 0)
+    {
+        execlp("tput", "tput", cap, param, NULL);
+        exit(EXIT_FAILURE);
+    }
+    waitpid(cpid, &status, 0);
+}
+
+/****************************************************************************************
+                    BSP CONSOLE IMPLEMENTATION FUNCTIONS
+ ****************************************************************************************/
+
+/*----------------------------------------------------------------
+   OS_BSP_ConsoleOutput_Impl
+   See full description in header
+ ------------------------------------------------------------------*/
+void OS_BSP_ConsoleOutput_Impl(const char *Str, uint32 DataLen)
+{
+    /* writes the raw data directly to STDOUT_FILENO (unbuffered) */
+    write(STDOUT_FILENO, Str, DataLen);
```

Comment:
/export/home/jhageman/cFS/cFS-GitHub/osal/src/bsp/pc-linux/src/bsp_console.c:68:5: error: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Werror=unused-result]
     write(STDOUT_FILENO, Str, DataLen);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Reply:
f59b16b puts this write in a loop that checks the return code.  There is still no action taken if it fails (there isn't much to do).